assets/js/segment.js

```diff
@@ -1,14 +1,40 @@
 window.addEventListener('DOMContentLoaded', function () {
+  function copy(node) {
+    let copyText;
+
+    if (navigator.clipboard) {
+      // Modern browser
+      copyText = node.textContent;
+      navigator.clipboard.writeText(copyText);
+    } else {
+      const selection = getSelection();
+      selection.removeAllRanges();
+
+      const range = document.createRange();
+      range.selectNodeContents(node);
+      selection.addRange(range);
+      copyText = range.toString();
+
+      document.execCommand('copy');
+      selection.removeAllRanges();
+    }
+
+    return copyText;
+  }
+
   // If target element is code copy
   function isTargetCodeCopyButton(elem) {
     return elem.classList.contains('fa-copy');
   }
 
   // Determine whether we should track or not
   function shouldTrack(e) {
-    if (e.target.tagName === 'A' || isTargetCodeCopyButton(e.target)) {
+    if (e.target.tagName === 'A') {
       return { target: e.target };
-    } else if (e.target.parentNode.tagName === 'A') {
+    } else if (
+      isTargetCodeCopyButton(e.target) ||
+      e.target.parentNode.tagName === 'A'
```

Comment:
Maybe not just parent,  we should check all top parents 
I suggest use:
`if(e.target.closest('a')){ ... }`

Reply:
I looked at the structure for the site, and it's pretty straightforward. I think using `closest` (it probably has some perf impact) could certainly work, but I'll have to refactor more to accommodate that use case. I'll address that at a separate point.